includes/discovery/storage/eql-storage.inc.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * LibreNMS
+ *
+ * Copyright (c) 2016 Peter TKATCHENKO https://github.com/Peter2121/
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+$eql_storage = snmpwalk_cache_oid($device, 'EqliscsiVolumeEntry', null, 'EQLVOLUME-MIB', $config['install_dir'].'/mibs/equallogic');
```

Comment:
Hi,

Two others (hrstorage and netapp) have no this check.
The only working line is snmpwalk_cache_oid(...) without this check.
Do you really think this check would be useful?
Are use sure $os is always populated when this include works?

Peter


On 05/17/2016 00:24, Neil Lathwood wrote:
>
> In includes/discovery/storage/eql-storage.inc.php 
> <https://github.com/librenms/librenms/pull/3519#discussion_r63436648>:
>
> > @@ -0,0 +1,37 @@
> > +<?php
> > +
> > +/*
> > + * LibreNMS
> > + *
> > + * Copyright (c) 2016 Peter TKATCHENKO https://github.com/Peter2121/
> > + *
> > + * This program is free software: you can redistribute it and/or modify it
> > + * under the terms of the GNU General Public License as published by the
> > + * Free Software Foundation, either version 3 of the License, or (at your
> > + * option) any later version.  Please see LICENSE.txt at the top level of
> > + * the source code distribution for details.
> > + */
> > +
> > +$eql_storage = snmpwalk_cache_oid($device, 'EqliscsiVolumeEntry', null, 'EQLVOLUME-MIB', $config['install_dir'].'/mibs/equallogic');
>
> Think you need to check if the os is equallogic first otherwise this 
> will run for all devices.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/librenms/librenms/pull/3519/files/c99c74bd9554ad24654cd690b5b337ed5da8f47c#r63436648>
>